### PR TITLE
[RFC] Modify vim_strbyte() to take advantage of strchr()

### PR DESCRIPTION
**Pull request by amaro**
ref #1476

I wanted to work on an entry-level issue to get acquainted to the code. This is my first time creating a PR so hopefully this hasn't been fixed and I did everything right.

@aktau mentioned there should be tests for this change. Concept-wise, they should be unit tests, but maybe the terminology is different for this project? I'll work on the tests anyway, just wanted to confirm the change looks decent. Are there any existing tests creating weird char buffers I can look at?

I created https://gist.github.com/amaro/d5ff69fd53e7904c00ad to convince myself this would work in all cases.

### CODE REVIEW

**Comment by amaro** [comment]
I forgot to include the function attributes. Should I just create a new commit on top of my branch?

**Comment by Pyrohh** [comment]
> I forgot to include the function attributes. Should I just create a new commit on top of my branch?

No need, this PR is quite small. You can just amend the existing commit.

File: src/nvim/strings.c [hunk 1]
```diff
@@ -413,21 +413,18 @@ char_u *vim_strchr(const char_u *string, int c)
 }
 
 /*
- * Version of strchr() that only works for bytes and handles unsigned char
- * strings with characters above 128 correctly. It also doesn't return a
- * pointer to the NUL at the end of the string.
+ * Search for c in buf. Appropriately handles values of c in [0, 255].
```
Comment (Pyrohh) [inline src/nvim/strings.c]: Please use Doxygen style comments `///`, and function parameters should be surrounded in backticks, e.g. ``` `c` ```.

File: src/nvim/strings.c [hunk 2]
```diff
@@ -413,21 +413,18 @@ char_u *vim_strchr(const char_u *string, int c)
...
+ * This function preserves backward compatibility with previous vim code.
```
Comment (Pyrohh) [inline src/nvim/strings.c]: I don't think this needs to be mentioned. We're documenting existing codebase, not the history, although I wonder what others think.

File: src/nvim/strings.c [hunk 3]
```diff
@@ -413,21 +413,18 @@ char_u *vim_strchr(const char_u *string, int c)
...
  */
-char_u *vim_strbyte(const char_u *string, int c)
-  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
-{
-  const char_u *p = string;
+char_u *vim_strbyte(const char_u *str, int c) {
+  /* if c == 0, then we are looking for the NUL char. The old version of this func
+   *   returned NULL in this case.
+   * if c < 0 or c > 255,  then we are looking for a byte outside the unsigned char range.
+   *   In this case, the old func returned NULL. */
```
Comment (Pyrohh) [inline src/nvim/strings.c]: same here regarding comment style, and I don't think the old function's behavior has any relevance

File: src/nvim/strings.c [hunk 4]
```diff
@@ -413,21 +413,18 @@ char_u *vim_strchr(const char_u *string, int c)
...
+  if (c > (unsigned char) -1 || c <= 0)
```
Comment (Pyrohh) [inline src/nvim/strings.c]: braces around if statement

File: src/nvim/strings.c (hunk 2 above)
Comment (justinmk) [inline src/nvim/strings.c]: Unless there is something special about this case (e.g. is this function at particular risk of being erroneously removed, compared to the hundreds of other such functions?), there is no need to mention compatibility.

**Comment by justinmk** [comment]
Besides unit tests, some performance results with actual vimscript (or searching via `/` or `search()`) is a large part of the work needed for #1476.

**Pull request closed without merging**